test/CMakeLists.txt

```diff
@@ -58,14 +62,14 @@ endif()
 enable_testing()
 
 # Note: doctest and similar testing frameworks can automatically configure CMake tests For other
-# testing frameworks add the tests target instead: ADD_TEST(GreeterTests GreeterTests)
-
-include(${doctest_SOURCE_DIR}/scripts/cmake/doctest.cmake)
-doctest_discover_tests(GreeterTests)
+# testing frameworks add the tests target instead:
+add_test(greeterTests GreeterTests)
+# include(${doctest_SOURCE_DIR}/scripts/cmake/doctest.cmake) doctest_discover_tests(GreeterTests)
+# Warning: if doctest is imported with find_package() this will fail! CK
```

Comment:
By default, Doctest should be always added from source here, safely continue using the `doctest_discover_tests` helper function.

Reply:
only on your CI

Reply:
It seems you have the `CPM_USE_LOCAL_PACKAGES` flag enabled? This is not recommended for library development, as your build wont be reliable, but dependent on your current system and configuration, as seen here.

Reply:
I like the **CPM-cmake** concept really!

But i.e., I do not want to build the **boost** or any other big library every time on my host while using it.

I use **brew** and if needed `find_package(Boost 1.75.1 EXACT)`